rosette/src/Export.cc

```diff
@@ -200,19 +192,28 @@ void letExprObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::Ob
 }
 
 void tupleExprObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::ObType pbtype ) {
-    ObjectCodePB::TupleExpr *teob = lvob->mutable_tupleexpr();
+    ObjectCodePB::TupleExpr *teob = lvob->mutable_tuple_expr();
     TupleExpr * te = (TupleExpr *)ob;
 
+    // There are cases where the TupleExpr references itself in the "rest" field. This could
```

Comment:
Can you tell me where and how you observed this?

Reply:
CodeId=1708
litvec:
   0:   {RequestExpr} (1690)
   1:   '**SOURCE-FILE** (1644)
   2:   "./boot.rbl" (1688)
codevec:
   0:   alloc 2
   1:   liti 1,arg[0]
   2:   liti 2,arg[1]
   3:   add 2,rslt
   5:   rtn/nxt

code_block {
  object_id: 1708
  litvec {
    ob {
      type: OT_REQUEST_EXPR
      object_id: 1690
      meta_id: 119
      parent_id: 120
      request_expr {
        target {
          type: OT_SYMBOL
          object_id: 1644
          meta_id: 308
          parent_id: 309
          symbol {
            name: "add"
          }
        }
        msg {
          type: OT_TUPLE_EXPR
          object_id: 1689
          meta_id: 1154
          parent_id: 108
          tuple_expr {
            rest {
              type: OT_TUPLE_EXPR
              object_id: 320
              meta_id: 1154
              parent_id: 108
              tuple_expr {
                rest {
                  object_id: 320
                  looped: true
                }
              }
            }
          }
        }
      }
    }
    ob {
      type: OT_SYMBOL
      object_id: 1644
      meta_id: 308
      parent_id: 309
      symbol {
        name: "**SOURCE-FILE**"
      }
    }
    ob {
      type: OT_RBL_STRING
      object_id: 1688
      meta_id: 23
      parent_id: 24
      rbl_string {
        value: "./boot.rbl"
      }
    }
  }
  codevec {
    opcodes: "\002\004\001\272\022\272\0028\000\243\000M0$nV\254\006\000\000\000\000\000\000"
  }
}


The "looped" field indicates that the "rest" pointer in the TupleExpr points to itself, i.e. the TupleExpr containing it.